From ccc263d1d9a58a0161d4ea98565deb9c59123b3a Mon Sep 17 00:00:00 2001
From: PratikBhusal <PratikBhusal@users.noreply.github.com>
Date: Wed, 24 Oct 2018 15:19:44 -0500
Subject: [PATCH] Init window configs from ~/.Xresources file

The following setitngs from config.h can now be loaded from your
Xresources config file:

BAR_FONT     ( default: sxiv.font )
WIN_BG_COLOR ( default: sxiv.background )
WIN_FS_COLOR ( default: sxiv.fsColor )
SEL_COLOR    ( default: sxiv.selColor )
BAR_BG_COLOR ( default: sxiv.barBackground )
BAR_FG_COLOR ( default: sxiv.barForeground )

Note: you do need to get a new copy of config.h as this patch modifies
config.def.h
---
 config.def.h | 26 ++++++++++++++++++++------
 window.c     |  2 ++
 2 files changed, 22 insertions(+), 6 deletions(-)

diff --git a/config.def.h b/config.def.h
index 58e728e..7d48c6c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -9,16 +9,30 @@ enum {
 /* bar font:
  * (see fonts-conf(5) subsection "FONT NAMES" for valid values)
  */
-static const char * const BAR_FONT = "monospace:size=8";
+static const char * BAR_FONT = "monospace:size=8";
 
 /* colors:
  * (see X(7) section "COLOR NAMES" for valid values)
  */
-static const char * const WIN_BG_COLOR = "#555555";
-static const char * const WIN_FS_COLOR = "#000000";
-static const char * const SEL_COLOR    = "#EEEEEE";
-static const char * const BAR_BG_COLOR = "#222222";
-static const char * const BAR_FG_COLOR = "#EEEEEE";
+static const char * WIN_BG_COLOR = "#555555";
+static const char * WIN_FS_COLOR = "#000000";
+static const char * SEL_COLOR    = "#EEEEEE";
+static const char * BAR_BG_COLOR = "#222222";
+static const char * BAR_FG_COLOR = "#EEEEEE";
+
+/*
+ * Xresources preference(s) for window settings to load at startup
+ */
+#ifdef _XRESOURCES_CONFIG
+    static const ResourcePref win_resources[] = {
+        { "font",          STRING, &BAR_FONT     },
+        { "background",    STRING, &WIN_BG_COLOR },
+        { "fsColor",       STRING, &WIN_FS_COLOR },
+        { "selColor",      STRING, &SEL_COLOR    },
+        { "barBackground", STRING, &BAR_BG_COLOR },
+        { "barForeground", STRING, &BAR_FG_COLOR },
+    };
+#endif /* _XRESOURCES_CONFIG */
 
 #endif
 #ifdef _IMAGE_CONFIG
diff --git a/window.c b/window.c
index 2ed33ca..03a58cd 100644
--- a/window.c
+++ b/window.c
@@ -122,6 +122,8 @@ void win_init(win_t *win)
 	if (setlocale(LC_CTYPE, "") == NULL || XSupportsLocale() == 0)
 		error(0, 0, "No locale support");
 
+	init_xresources(e->dpy, win_resources, ARRLEN(win_resources));
+
 	win_init_font(e, BAR_FONT);
 
 	win_alloc_color(e, WIN_BG_COLOR, &win->bgcol);
-- 
2.17.1

